File: 1-Two Sum/TwoSum.cpp
#include <iostream>
#include <unordered_map>
#include <vector>

using namespace std;


vector<int> twoSum(vector<int> &list , int target) {
    unordered_map<int , int> map;
    vector<int> output;
    for (int i = 0 ; i < list.size() ; i++) {
        int x = target - list[i];
        if(map.find(x) != map.end()) {
            return {map[x],i};
        }
        map[list[i]] = i;
    }

    return {};
}




int main()
{

    vector<int> arr = {7,6,3,7,9,12,34,5,7};
    vector<int> rslt = twoSum(arr,14);

    cout << rslt[0] << rslt[1] ;
    
    return 0;
}

File: 41-Two Sum III - Data structure design/TwoSum.py
class TwoSum :
    def __init__(self):
        self.data = []
    
    def add(self ,x : int ) :
         self.data.append(x)
    
    
    def find(self , x : int ) -> bool :
        s = set()
        for num in self.data:
            y = x - num
            if y in s:
                return True
            s.add(num)
            
        return False
            
[1,3,5,7] = 4